ops/model.py

```diff
@@ -1279,6 +1300,7 @@ def __init__(
         self._id = id
         self._label = label
         self._content = content
+        self._secret_set_cache = _secret_set_cache or collections.defaultdict(dict)
```

Comment:
> Wouldn't this have to be an `is None` check, otherwise if the passed-in cache dict is empty it'll evaluate to false here and always create a new dict.

Yes, good catch, thanks!

> I guess we could make it a required argument after `backend` instead, and make it non-optional?

Although the docs say

> This class should not be instantiated directly

that would still break backwards compatibility, so I'd prefer to not do that.